vizro-core/examples/_dev/app.py

```diff
@@ -2,32 +2,44 @@
 
 import vizro.models as vm
 import vizro.plotly.express as px
+from flask_caching import Cache
 from vizro import Vizro
-from vizro.tables import dash_ag_grid
+from vizro.managers import data_manager
 
-df = px.data.gapminder()
+df = px.data.iris()
+
+# data_manager.cache = Cache(config={"CACHE_TYPE": "SimpleCache", "CACHE_DEFAULT_TIMEOUT": 20})
+data_manager.cache = Cache(config={"CACHE_TYPE": "FileSystemCache", "CACHE_DIR": "cache", "CACHE_DEFAULT_TIMEOUT": 20})
+# I didn't test this one yet:
+# data_manager.cache = Cache(config={"CACHE_TYPE": "RedisCache"})
+data_manager["default_expire_data"] = lambda: px.data.iris()
+
+# Set cache of fast_expire_data to expire every 10 seconds
+data_manager["fast_expire_data"] = px.data.iris()
+# data_manager["fast_expire_data"].timeout = 5
+# Set cache of no_expire_data to never expire
+data_manager["no_expire_data"] = lambda: px.data.iris()
+data_manager["no_expire_data"].timeout = 0
 
 page = vm.Page(
-    title="Enhanced AG Grid",
+    title="Blah",
     components=[
-        vm.AgGrid(
-            title="Dash AG Grid",
-            figure=dash_ag_grid(
-                data_frame=df,
-                columnDefs=[
-                    {"field": "country", "floatingFilter": True, "suppressHeaderMenuButton": True},
-                    {"field": "continent", "floatingFilter": True, "suppressHeaderMenuButton": True},
-                    {"field": "year"},
-                    {"field": "lifeExp", "cellDataType": "numeric"},
-                    {"field": "pop", "cellDataType": "numeric"},
-                    {"field": "gdpPercap", "cellDataType": "euro"},
-                ],
-            ),
-        ),
+        vm.Graph(figure=px.scatter(df, "sepal_width", "sepal_length")),
+        vm.Graph(figure=px.scatter(df, "sepal_width", "sepal_length")),
+        vm.Graph(figure=px.scatter("default_expire_data", "sepal_width", "sepal_length")),
+        vm.Graph(figure=px.scatter("fast_expire_data", "sepal_width", "sepal_length")),
+        vm.Graph(figure=px.scatter("no_expire_data", "sepal_width", "sepal_length")),
     ],
-    controls=[vm.Filter(column="continent")],
+    controls=[vm.Filter(column="species")],
 )
 dashboard = vm.Dashboard(pages=[page])
+app = Vizro().build(dashboard)
+server = app.dash.server
 
 if __name__ == "__main__":
-    Vizro().build(dashboard).run()
+    app.run()
+    # app.run(processes=2, threaded=False)
+    # gunicorn -w 2 app:server -b localhost:8050
+    # should also work with --preload
```

Comment:
Yeah, I also noticed this and was confused by it. It's to do with how [`run_simple`](https://werkzeug.palletsprojects.com/en/3.0.x/serving/#werkzeug.serving.run_simple), which is ultimately what Flask's development server and hence `app.run` uses, works. Unlikely gunicorn which forks two long-lived processes and reuses them, werkzeug forks a new short-lived process every request. The significance of the `processes=2` is the number of processes it can concurrently fork I guess. I don't know the details of the mechanism, but basically it's because it's just a development server and not a "proper" way of handling multiple processes.

Reply:
> Unlikely gunicorn which forks two long-lived processes and reuses them, werkzeug forks a new short-lived process every request. The significance of the processes=2 is the number of processes it can concurrently fork I guess 

From what I can see from the documentation you mentioned, this appears to be true.